File: aoc24/lea/1.js
const fs = require("fs");

let left = [];
let right = [];

const readInput = (filepath) => {
  const file = fs.readFileSync(filepath, "utf-8");
  file.split(/\r?\n/).forEach((line, index) => {
    res = line.split(/(\s+)/);
    left.push(res[0]);
    right.push(res[res.length - 1]);
  });
};

const part1 = (left, right) => {
  left.sort();
  right.sort();

  let sumd = 0;

  for (let i = 0; i < left.length; i++) {
    sumd += Math.abs(left[i] - right[i]);
  }
  return sumd;
};

const part2 = (left, right) => {
    right.sort();

    let sumM = 0;
    
    left.forEach(f => {
        let amount = right.filter(s => s === f).length;
        sumM += f * amount;
    });

    return sumM;
}

readInput("./1-puzzle.txt");

console.log(part1(left, right));
console.log(part2(left, right));

File: aoc25/lea/1/1.py
# PART1
c1 = 0
curr = 50
with open("1-input.txt", encoding="utf-8") as f:
    for line in f:
        distance = int(line[1:])
        if "L" in line:
            curr = (curr - distance) % 100
        else:
            curr = (curr + distance) % 100
        if curr == 0:
            c1 += 1
print("part1:", c1)

# PART2
curr = 50
c2 = 0
with open("1-input.txt", encoding="utf-8") as f:
    for line in f:
        distance = int(line[1:])

        if "L" in line:
            if curr - distance < 0:
                c2 += abs((100 + curr - distance) // 100)
                if curr != 0:
                    c2 += 1
            curr = (curr - distance) % 100
            if curr == 0:
                c2 += 1
        else:
            c2 += (curr + distance) // 100
            curr = (curr + distance) % 100

print("part2:", c2)
